cmd/optimizer-nri-plugin/main.go

```diff
@@ -172,37 +197,127 @@ func (p *plugin) StartContainer(_ *api.PodSandbox, container *api.Container) err
 		persistFile = fmt.Sprintf("%s.timeout%ds", persistFile, cfg.Timeout)
 	}
 
-	fanotifyServer := fanotify.NewServer(cfg.ServerPath, container.Pid, imageName, persistFile, cfg.Readable, cfg.Overwrite, time.Duration(cfg.Timeout)*time.Second, logWriter)
+	var hasSentPrefetchList = false
+
+	fanotifyServer := fanotify.NewServer(cfg.ServerPath, container.Pid, imageName, persistFile, cfg.Readable, cfg.Overwrite, time.Duration(cfg.Timeout)*time.Second, logWriter, containerName, hasSentPrefetchList)
 
 	if err := fanotifyServer.RunServer(); err != nil {
 		return err
 	}
 
+	go func() {
+		time.Sleep(10 * time.Minute)
+		fanotifyServer.Mu.Lock()
+		if !fanotifyServer.IsSent {
+			data, err := getPrefetchList(persistFile)
+			if err != nil {
+				log.WithError(err).Error("error reading file")
+			}
+			if err = sendToServer(imageRepo, containerName, cfg.PrefetchDistributionEndpoint, data); err != nil {
+				log.WithError(err).Error("failed to send prefetch to http server")
+			}
+			fanotifyServer.IsSent = true
+		}
+		fanotifyServer.Mu.Unlock()
+	}()
+
 	globalFanotifyServer[imageName] = fanotifyServer
 
 	return nil
 }
 
+func sendToServer(imageName, containerName, serverURL string, data []byte) error {
+	filePaths := strings.Split(string(data), "\n")
+
+	var prefetchFiles []PrefetchFile
+	for _, path := range filePaths {
+		if path != "" {
+			prefetchFiles = append(prefetchFiles, PrefetchFile{Path: path})
+		}
+	}
+
+	item := CacheItem{
```

Comment:
Can we define the generic API for different storage service? Sometimes, we may need to send more information to the server.

Reply:
Also, does the API provide a query endpoint?